src/main/scala/is/hail/expr/Type.scala

```diff
@@ -689,9 +707,10 @@ abstract class TContainer extends Type {
     region.invoke[Long, Int]("loadInt", aoff)
 
   def isElementDefined(region: MemoryBuffer, aoff: Long, i: Int): Boolean =
-    !region.loadBit(aoff + 4, i)
+    elementsRequired || !region.loadBit(aoff + 4, i)
 
   def setElementMissing(region: MemoryBuffer, aoff: Long, i: Int) {
+    assert(!elementsRequired, "element is required")
```

Comment:
:stop_sign: 
this'll allocate!

Reply:
(see the discussion last week about asserts, Scala is :1st_place_medal: )

Reply:
womp >.< forgot to take that out